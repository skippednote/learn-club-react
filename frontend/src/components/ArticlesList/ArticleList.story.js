import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';
import ArticleList from './index';
import fixture from '../../pages/Articles/Articles.fixture';

storiesOf('ArticleList', module).add('Default', () => (
  <MemoryRouter>
    <ArticleList articles={fixture} loading={false} error={null} />
  </MemoryRouter>
));
