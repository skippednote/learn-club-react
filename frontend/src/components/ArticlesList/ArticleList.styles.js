import styled from 'react-emotion';

export const List = styled('ul')`
  padding-left: 0;
  list-style: none;
`;
List.displayName = 'ArticleList';

export const Item = styled('li')`
  text-align: left;
  border-bottom: 1px solid #ddd;
  margin-bottom: 25px;
  padding-bottom: 25px;
`;
Item.displayName = 'ArticleItem';

export const More = styled('div')`
  margin-top: 10px;
`;
More.displayName = 'ArticleMore';
