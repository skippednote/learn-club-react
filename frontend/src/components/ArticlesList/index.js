// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import type { Article } from '../../pages/Articles/Articles.types';
import { List, Item, More } from './ArticleList.styles';

type Props = {
  articles: Article
};

class ArticleList extends Component<Props> {
  render() {
    const {
      articles: { data }
    } = this.props;
    return (
      <List className="articlelist">
        {data.map(({ id, attributes: { title, body } }) => (
          <Item key={id}>
            <h2>{title}</h2>
            <div dangerouslySetInnerHTML={{ __html: body.summary }} />
            <More>
              <Link to={`/articles/${id}`}>Read more</Link>
            </More>
          </Item>
        ))}
      </List>
    );
  }
}

export default ArticleList;
