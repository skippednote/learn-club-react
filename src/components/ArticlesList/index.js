// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import type { Articles } from '../../pages/Articles/Articles.types';
import { List, Item, More } from './ArticleList.styles';
import StateMessage from '../StateMessage';

type Props = {
  articles: Articles
};

class ArticleList extends Component<Props> {
  render() {
    const {
      articles: { nodeArticle }
    } = this.props;
    const articleIDs = nodeArticle && Object.keys(nodeArticle);

    return (
      <List className="articlelist">
        {articleIDs ? (
          articleIDs.map(articleID => (
            <Item key={articleID}>
              <h2>{nodeArticle[articleID].attributes.title}</h2>
              <div
                dangerouslySetInnerHTML={{
                  __html: nodeArticle[articleID].attributes.body.summary
                }}
              />
              <More>
                <Link to={`/articles/${articleID}`}>Read more</Link>
              </More>
            </Item>
          ))
        ) : (
          <StateMessage state="loading">No Articles found!</StateMessage>
        )}
      </List>
    );
  }
}

export default ArticleList;
