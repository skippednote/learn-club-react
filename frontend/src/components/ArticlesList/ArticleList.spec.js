import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ArticlesList from './index';
import fixture from '../../pages/Articles/Articles.fixture';

const props = {
  articles: fixture,
  error: null,
  loading: false
};

describe('ArticlesList', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ArticlesList {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders a list of articles', () => {
    const wrapper = shallow(<ArticlesList {...props} />);
    const articleItem = wrapper.find('ArticleItem');
    expect(articleItem).toHaveLength(2);
  });
});
