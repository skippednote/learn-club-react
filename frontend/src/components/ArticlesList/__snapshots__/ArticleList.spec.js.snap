// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList matches snapshot 1`] = `
<ArticleList
  className="articlelist"
>
  <ArticleItem
    key="e4e23b35-468d-4f6a-ade5-df1b375dec0d"
  >
    <h2>
      Ergo ita: non posse honeste vivi, nisi honeste vivatur
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cur igitur, cum de re conveniat, non malumus usitate loqui? Quis enim redargueret? Illa enim, quae prosunt aut quae nocent, aut bona sunt aut mala, quae sint paria necesse est. Tu enim ista lenius, hic Stoicorum more nos vexat. Duo Reges: constructio interrete. Et hunc idem dico, inquieta sed ad virtutes et ad vitia nihil interesse. Omnis sermo elegans sumi potest, tum varietas est tanta artium, ut nemo sine eo instrumento ad ullam rem illustriorem satis ornatus possit accedere. De malis autem et bonis ab iis animalibus, quae nondum depravata sint, ait optime iudicari. Quid in isto egregio tuo officio et tanta fide-sic enim existimo-ad corpus refers? Semovenda est igitur voluptas, non solum ut recta sequamini, sed etiam ut loqui deceat frugaliter. Hanc se tuus Epicurus omnino ignorare dicit quam aut qualem esse velint qui honestate summum bonum metiantur. Mihi vero, inquit, placet agi subtilius et, ut ipse dixisti, pressius. Longum est enim ad omnia respondere, quae a te dicta sunt.",
        }
      }
    />
    <ArticleMore>
      <Link
        replace={false}
        to="/articles/e4e23b35-468d-4f6a-ade5-df1b375dec0d"
      >
        Read more
      </Link>
    </ArticleMore>
  </ArticleItem>
  <ArticleItem
    key="e4e23b35-468d-4f6a-ade5-df1b375dec0d"
  >
    <h2>
      Cur igitur, cum de re conveniat, non malumus usitate loqui?
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cur igitur, cum de re conveniat, non malumus usitate loqui? Quis enim redargueret? Illa enim, quae prosunt aut quae nocent, aut bona sunt aut mala, quae sint paria necesse est. Tu enim ista lenius, hic Stoicorum more nos vexat. Duo Reges: constructio interrete. Et hunc idem dico, inquieta sed ad virtutes et ad vitia nihil interesse. Omnis sermo elegans sumi potest, tum varietas est tanta artium, ut nemo sine eo instrumento ad ullam rem illustriorem satis ornatus possit accedere. De malis autem et bonis ab iis animalibus, quae nondum depravata sint, ait optime iudicari. Quid in isto egregio tuo officio et tanta fide-sic enim existimo-ad corpus refers? Semovenda est igitur voluptas, non solum ut recta sequamini, sed etiam ut loqui deceat frugaliter. Hanc se tuus Epicurus omnino ignorare dicit quam aut qualem esse velint qui honestate summum bonum metiantur. Mihi vero, inquit, placet agi subtilius et, ut ipse dixisti, pressius. Longum est enim ad omnia respondere, quae a te dicta sunt.",
        }
      }
    />
    <ArticleMore>
      <Link
        replace={false}
        to="/articles/e4e23b35-468d-4f6a-ade5-df1b375dec0d"
      >
        Read more
      </Link>
    </ArticleMore>
  </ArticleItem>
</ArticleList>
`;
